.github/workflows/ci.yml

```diff
@@ -13,19 +13,63 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python 3.7
+        id: setup-python
         uses: actions/setup-python@v2
         with:
           python-version: 3.7
-      - name: Install Flair dependencies
-        run: pip install -e .
-      - name: Install unittest dependencies
-        run: pip install -r requirements-dev.txt
-      - name: Show installed dependencies
-        run: pip freeze
+      #----------------------------------------------
+      #       Try to load cached poetry installation
+      #----------------------------------------------
+      - name: Load cached Poetry installation
```

Comment:
This caching mechanism looks rather complex,

`actions/setup-python` provides that functionality out of the box, see [their caching section](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages)

Reply:
It follows https://github.com/marketplace/actions/install-poetry-action - it looks like setup-python has something that is similar to the "cached venv" part, but does not seem to cover the "cached poetry install" part.
If we wish to prioritize a simpler ci.yaml we can use this, but it will be a bit slower, as it will always install poetry.